code/contracts/ONEWallet.sol

```diff
@@ -263,7 +271,7 @@ contract ONEWallet {
         nonces[index] = v + 1;
     }
 
-    // same output as abi.encodePacked(x), but the behavior of abi.encodePacked is sort of uncertain
+    // same output as abi.encodePacked(x), but the behavior of abi.encodePacked is sort of uncertain - IH: this is surprising; in which sense?
```

Comment:
Based on the spec here https://docs.soliditylang.org/en/latest/abi-spec.html#non-standard-packed-mode , arrays may get padded with `abi.encodePacked`.  This is not the case here, but I do not want to rely on this function since its behavior may change in the future. Our use case is pretty simple: given a list of parameters, convert them into 32-byte elements and concatenate them together. The whole `abi` encoding spec seems to be an overkill for this use case.

Reply:
I am sure that its behavior will not change. If anything like that would be about to happen, not in the same version of solidity, so it would not impact us. This function provides unambiguous concatenation and I've seen it in many other contracts.

Reply:
Let's stick to simpler expression on both contract and Javascript sides for now. The Javascript side of the code for abi.encodePacked looks quite messy.